Library/Homebrew/tap.rb

```diff
@@ -280,4 +305,70 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
```

Comment:
> I really think it's bad smell (by reinventing the wheel).

And I really think using a Singleton is a bad smell but I'm willing to compromise.



> Logically speaking, this is totally irrelevant on whether should we use it here.

It's not totally irrelevant; it adds weight to the argument that "we don't do things this way in Homebrew core". 

Really, though, do what you want. I'm trying to compromise here to reach some sort of middle ground but it appears you're unwilling to do the same. If you'd rather, I'll just stopping reviewing this PR and you can let someone else do it.